cognite/client/_api/datapoints.py

```diff
@@ -1026,6 +1026,11 @@ def retrieve_dataframe_in_tz(
             3. Create a query for each interval and pass all to the retrieve_arrays method.
             4. Stack the resulting arrays into a single column in the resulting DataFrame.
 
+        The splitting of time range into UTC intervals can lead to ambiguous timestamps. An ambiguous timestamp is,
+        for example, 1916-10-01 00:00:00 Europe/Oslo as this have two UTC times 1916-10-01 00:00:00+02:00
+        and 1916-10-01 00:00:00+01:00 depending on whether you are before or after the DST transition. In this case,
+        the splitting always chose the DST.
+
```

Comment:
I vote we just remove this for 2 reasons:

1. The error did not arise because of a user input error (start/end)...
2. ...as it is not possible to specify an ambiguous `datetime` (as far as I understand) since `datetime` uses `fold=0` by default (i.e. pick "first" / summer time). When/if a user would supply us with the same `datetime` but with `fold=1` it's also not a problem: With granularity of hour and below, we know when to start (pick "last" / winter time); with granularity of day and above, we always floor it, and the result of this is not ambiguous.

Maybe there's something I haven't considered 😅 

```suggestion
```

Reply:
I think you are right. I originally thought of this as just a choice we make (which timestamp to chose), both being correct. And you are right there is a correct one, the one at the beginning of the month. Unfortunately, this is not what the code is always doing. Currently we are always choosing DST time, which is correct in the fall, but in the spring that means we lose one hour of the month. 

Can you check my thinking here? If it is true I might have to reconsider the implementation of the fix.

Reply:
Actually, you don't - the one in spring is never ambiguous, but it can represent a non-existent time. I see that pandas provides the following options:
```
A nonexistent time does not exist in a particular timezone where clocks moved forward due to DST.
- ‘shift_forward’ will shift the nonexistent time forward to the closest existing time
- ‘shift_backward’ will shift the nonexistent time backward to the closest existing time
- ‘NaT’ will return NaT where there are nonexistent times
- timedelta objects will shift nonexistent times by the timedelta
- ‘raise’ will raise an NonExistentTimeError if there are nonexistent times.
```

I wonder if it is as simple as providing `"shift_backward"`, but it's to early in the morning for me to think this through 😂

Reply:
Found this: https://stackoverflow.com/a/18490073/3887338

Will play around with one of these that do not have "midnight".

Reply:
Not sure if I follow you here. When you say `'Actually, you don't'`, you mean the spring one is not a problem as it is not in a month transition? 

The `nonexistent` as far as I understand it, is for the opposite problem, when you instead of having two UTC timestamps for a local time, it is when you try to use a local time that does not exists due to it is in a DST transition.

Anyways, so solution is do delete the above comment?

Reply:
After some thinking: `"shift_forward"` is correct:

![Screenshot 2023-04-27 at 09 55 49](https://user-images.githubusercontent.com/8521241/234802055-781f3801-26aa-4ee0-8f05-e053b1e1558d.png)

Reply:
Updated it now. Hopefully good.